### PR TITLE
Add Exclude support in select

### PR DESCRIPTION
**Pull request by mustafasrepo**
# Which issue does this PR close?

Closes [#6439](https://github.com/apache/arrow-datafusion/issues/6439).

# Rationale for this change
With this PR we can run `SELECT` queries with `EXCLUDE` where all the columns except `EXCLUDE` columns are projected, such as below.
```sql
SELECT * EXCLUDE(a, b)
  FROM table1
```

# What changes are included in this PR?

# Are these changes tested?
Yes new tests are added `select.slt` file, to check whether query produces expected columns.

# Are there any user-facing changes?

### CODE REVIEW

**Review by ozankabak — COMMENTED** [review]
A few questions except my in-line remarks:
1. Snowflake and DuckDB use the `EXCLUDE` token, BigQuery uses `EXCEPT` token. Can we support both?
2. Take a look at [Snowflake's syntax](https://docs.snowflake.com/en/sql-reference/sql/select). If one wants to exclude only one column, parenthesis are not required. Can we support this, or does it require `sqlparser` support? Let's support it if we can. If parse support is necessary, let's create a small PR on that repo and add DF side as a follow-on.

File: datafusion/core/tests/sqllogictests/test_files/select.slt [hunk 1]
```diff
@@ -759,6 +759,30 @@ SELECT a FROM annotated_data_finite2
 0
 0
 
+# below query should emit all the columns except a
```
[the remark this answers is not in the document]
Comment (mustafasrepo) [reply datafusion/core/tests/sqllogictests/test_files/select.slt:762]: Yes fixed it

File: datafusion/core/tests/sqllogictests/test_files/select.slt [hunk 2]
```diff
@@ -759,6 +759,30 @@ SELECT a FROM annotated_data_finite2
...
+        LIMIT 5
+----
+1 0 0
+1 1 2
+1 2 0
+1 3 0
+1 4 1
+
+# when wildcard is prepended with table name, exclude should still work
+# below query should emit all the columns except a
```
[the remark this answers is not in the document]
Comment (mustafasrepo) [reply datafusion/core/tests/sqllogictests/test_files/select.slt:775]: Yes, fixed it

**Comment by mustafasrepo** [comment]
Merged upstream

**Pull request closed without merging**